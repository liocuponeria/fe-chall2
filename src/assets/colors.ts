const red = '#FF0000'
const persianRose = '#F91E96'
const persianRose2 = '#F62996'
const forestGreen = '#24C92B'
const fruitSalad = '#4CAF50'
const japaneseLaurel = '#099F0F'

const black26 = '#00000026'
const black29 = '#00000029'
const blackB3 = '#000000B3'
const white = '#ffffff'

export const colors = Object.freeze({
  primary: blackB3,
  background: white,
  header: {
    left: red,
    right: persianRose,
  },
  navbar: {
    selected: red,
    shadow: black29,
  },
  sortBy: {
    border: persianRose2,
    divider: persianRose,
  },
  products: {
    border: black26,
    button: {
      left: forestGreen,
      right: japaneseLaurel,
      rover: japaneseLaurel,
    },
    details: {
      price: fruitSalad,
    },
  },
  footer: {
    left: red,
    right: persianRose,
  },
})
