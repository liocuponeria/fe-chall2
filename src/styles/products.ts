import styled from 'styled-components';
import { shade } from 'polished';

export const ContainerProduct = styled.div`
  max-width: 1070px;
  width: 100%;
  margin: auto;
  justify-content: space-around;
  display: flex;
  flex-wrap: wrap;
  @media screen and (min-width: 320px) and (max-width: 663px) {
    justify-content: center;
  }
  @media screen and (min-width: 664px) and (max-width: 772px) {
    justify-content: space-around;
  }
  .product {
    margin-top: 28px;
    border: 1px solid #00000026;
    border-radius: 14px;
    display: block;
    color: red;
    align-items: center;
    background: var(--unnamed-color-ffffff) 0% 0% no-repeat padding-box;
    box-shadow: 0px 6px 12px #0000001f;
    border: 1px solid #00000026;
    opacity: 1;
    transition: transform 0.2s;
    &:hover {
      transform: translateX(2px);
    }
  }
  img {
    display: flex;
    margin: 3px auto;
    width: 328px;
    height: 203px;
    max-width: 90%;
    background: transparent
      url('img/Captura de Tela 2020-12-08 Ã s 15.59.48.png') 0% 0% no-repeat
      padding-box;
    border-radius: 14px 14px 0 0;
  }
  h4 {
    max-width: 300px;
    height: 40px;
    overflow: hidden;
    justify-content: center;
    display: flex;
    margin: 12px auto;
    font-family: 'Roboto';
    font-size: 16px;
    font-weight: 400;
    color: #000000b3;
    opacity: 1;
  }
  p {
    text-align: center;
    margin-top: 5px;
    font-size: 22px;
    color: #000000b3;
    font-family: 'Roboto';
    letter-spacing: 0px;
    opacity: 1;
    font-weight: 700;
  }
  button {
    justify-content: center;
    align-items: center;
    font-size: 16px;
    font-family: 'Roboto';
    font-weight: bold;
    width: 302px;
    height: 36px;
    background: transparent linear-gradient(90deg, #24C92B 0%, #099F0F 100%) 0% 0% no-repeat padding-box;
    border: 0;
    font-wei
    border-radius: 18px;
    opacity: 1;
    display: flex;
    margin: 19px 13px 27px 13px;
    color: #fff;
    border-radius: 180px;
    box-shadow: 0px 3px 6px #00000029;
    &:hover {
      background: ${shade(0.2, '#24c92b')};
    }
  }
`;