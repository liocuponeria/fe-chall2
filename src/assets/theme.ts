import { colors } from '../assets/colors'

import { createGlobalStyle } from 'styled-components'

export const GlobalStyle = createGlobalStyle`
  body {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    -webkit-font-smoothing: antialiased !important;
    color: ${colors.primary};
    font-family: Roboto;
    background-color: ${colors.background};
  }
`

export const theme = {
  colors,
}
