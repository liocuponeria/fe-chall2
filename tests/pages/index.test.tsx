import renderer, { act } from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import IndexPage, { getStaticProps } from 'pages/index'
import { theme } from 'assets/theme'

import { mockProducts } from '../mock/products'

jest.mock('next/dist/client/router', () => ({
  useRouter: jest.fn().mockImplementation(() => ({
    query: {
      category: undefined,
    },
  })),
}))

jest.mock('hooks/filter', () => ({
  useFilter: jest.fn().mockImplementation(() => ({
    filter: 'empty',
    setFilter: jest.fn(),
  })),
}))

jest.mock('service/api', () => ({
  api: {
    get: jest.fn().mockImplementation(() => ({
      data: mockProducts,
    })),
  },
}))

describe('Index page', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <IndexPage products={mockProducts} />
      </ThemeProvider>
    )

    expect(tree.toJSON()).toMatchSnapshot()

    act(() => {
      tree.root.findByProps({ 'data-test-id': 'selectBoxContainer' }).props.onClick()
      tree.root.findByProps({ 'data-test-id': 'selectBoxOption0' }).props.onClick()
    })
  })

  it('getStaticProps', async () => {
    expect(await getStaticProps({})).toEqual({ props: { products: mockProducts } })
  })
})
