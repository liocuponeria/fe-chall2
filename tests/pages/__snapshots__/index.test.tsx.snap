// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page should match the snapshot 1`] = `
<div
  className="Background__BackgroundContainer-sc-1ig5col-0 caskQE"
>
  <div
    className="Header__HeaderContainer-sc-1a8uskq-0 iAAsmi"
  >
    <div>
      <img
        alt="Cuponeria Logo"
        src="/imgs/logo.svg"
      />
    </div>
  </div>
  <div
    className="NavBar__NavBarContainer-sc-8wct3k-0 jtmBiI"
  >
    <div>
      <div
        className="ButtonNavBar__ButtonNavBarContainer-sc-109tumo-0 hmJHDC"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <i
          className="icon-inicio"
        />
        <span>
          início
        </span>
      </div>
      <div
        className="ButtonNavBar__ButtonNavBarContainer-sc-109tumo-0 eVfgCa"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <i
          className="icon-roupas"
        />
        <span>
          roupas
        </span>
      </div>
      <div
        className="ButtonNavBar__ButtonNavBarContainer-sc-109tumo-0 eVfgCa"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <i
          className="icon-eletronicos"
        />
        <span>
          eletrônicos
        </span>
      </div>
      <div
        className="ButtonNavBar__ButtonNavBarContainer-sc-109tumo-0 eVfgCa"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <i
          className="icon-acessorios"
        />
        <span>
          acessórios
        </span>
      </div>
    </div>
  </div>
  <div
    className="childrenMargin"
  >
    <div
      className="SortBy__SortByContainer-akzcy7-0 jCamfy"
    >
      <div
        className="TitleContainer"
      >
        <span>
          Sugestões para você
        </span>
      </div>
      <div
        className="SelectBoxContainer"
      >
        <div
          className="SelectBox__SelectBoxContainer-r0v0pb-0 jGhjlW"
          data-test-id="selectBoxContainer"
          onClick={[Function]}
        >
          <div
            className="selectBoxPlaceholder"
          >
            ordenar por
          </div>
          <i
            className="icon-down-open"
          />
        </div>
        <div
          className="SelectBox__SelectBoxOptionsContainer-r0v0pb-1 iwxkET"
        >
          <div
            className="SelectBox__Option-r0v0pb-2 duzpYf optionContainer"
            data-test-id="selectBoxOption0"
            onClick={[Function]}
          >
            menor preço
          </div>
          <div
            className="divider"
          />
          <div
            className="SelectBox__Option-r0v0pb-2 duzpYf optionContainer"
            data-test-id="selectBoxOption1"
            onClick={[Function]}
          >
            a - z
          </div>
          <div
            className="divider"
          />
          <div
            className="SelectBox__Option-r0v0pb-2 duzpYf optionContainer"
            data-test-id="selectBoxOption2"
            onClick={[Function]}
          >
            mais recentes
          </div>
        </div>
      </div>
    </div>
    <div
      className="Products__ProductsContainer-sc-1itopdo-0 cQRWsy"
    >
      <div
        className="Product__ProductContainer-ih45b-0 dAlTtk"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="prodImg"
        />
        <div
          className="prodInfoContainer"
        >
          <span
            className="prodShortLabel"
          >
            Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
          </span>
          <span
            className="prodPrice"
          >
            R$ 
            55.99
          </span>
          <a
            className="BuyButton__BuyButtonContainer-sc-1fz6kqp-0 XtTgO"
          >
            Ver Produto
          </a>
        </div>
      </div>
      <div
        className="Product__ProductContainer-ih45b-0 iczRBy"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="prodImg"
        />
        <div
          className="prodInfoContainer"
        >
          <span
            className="prodShortLabel"
          >
            Mens Casual Premium Slim Fit T-Shirts 
          </span>
          <span
            className="prodPrice"
          >
            R$ 
            22.30
          </span>
          <a
            className="BuyButton__BuyButtonContainer-sc-1fz6kqp-0 XtTgO"
          >
            Ver Produto
          </a>
        </div>
      </div>
      <div
        className="Product__ProductContainer-ih45b-0 lfaRth"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="prodImg"
        />
        <div
          className="prodInfoContainer"
        >
          <span
            className="prodShortLabel"
          >
            Mens Cotton Jacket
          </span>
          <span
            className="prodPrice"
          >
            R$ 
            55.99
          </span>
          <a
            className="BuyButton__BuyButtonContainer-sc-1fz6kqp-0 XtTgO"
          >
            Ver Produto
          </a>
        </div>
      </div>
      <div
        className="Product__ProductContainer-ih45b-0 kmdYcv"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="prodImg"
        />
        <div
          className="prodInfoContainer"
        >
          <span
            className="prodShortLabel"
          >
            Mens Cotton Jacket
          </span>
          <span
            className="prodPrice"
          >
            R$ 
            15.99
          </span>
          <a
            className="BuyButton__BuyButtonContainer-sc-1fz6kqp-0 XtTgO"
          >
            Ver Produto
          </a>
        </div>
      </div>
      <div
        className="Product__ProductContainer-ih45b-0 dlSzSz"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="prodImg"
        />
        <div
          className="prodInfoContainer"
        >
          <span
            className="prodShortLabel"
          >
            John Hardy Women's Legends Naga Gold & Silver Dragon Station Chai...
          </span>
          <span
            className="prodPrice"
          >
            R$ 
            695.00
          </span>
          <a
            className="BuyButton__BuyButtonContainer-sc-1fz6kqp-0 XtTgO"
          >
            Ver Produto
          </a>
        </div>
      </div>
      <div
        className="Product__ProductContainer-ih45b-0 bezwcu"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="prodImg"
        />
        <div
          className="prodInfoContainer"
        >
          <span
            className="prodShortLabel"
          >
            Solid Gold Petite Micropave 
          </span>
          <span
            className="prodPrice"
          >
            R$ 
            168.00
          </span>
          <a
            className="BuyButton__BuyButtonContainer-sc-1fz6kqp-0 XtTgO"
          >
            Ver Produto
          </a>
        </div>
      </div>
      <div
        className="Product__ProductContainer-ih45b-0 dohCbi"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="prodImg"
        />
        <div
          className="prodInfoContainer"
        >
          <span
            className="prodShortLabel"
          >
            White Gold Plated Princess
          </span>
          <span
            className="prodPrice"
          >
            R$ 
            9.99
          </span>
          <a
            className="BuyButton__BuyButtonContainer-sc-1fz6kqp-0 XtTgO"
          >
            Ver Produto
          </a>
        </div>
      </div>
      <div
        className="Product__ProductContainer-ih45b-0 jEuIko"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="prodImg"
        />
        <div
          className="prodInfoContainer"
        >
          <span
            className="prodShortLabel"
          >
            Pierced Owl Rose Gold Plated Stainless Steel Double
          </span>
          <span
            className="prodPrice"
          >
            R$ 
            10.99
          </span>
          <a
            className="BuyButton__BuyButtonContainer-sc-1fz6kqp-0 XtTgO"
          >
            Ver Produto
          </a>
        </div>
      </div>
      <div
        className="Product__ProductContainer-ih45b-0 dlkOsT"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="prodImg"
        />
        <div
          className="prodInfoContainer"
        >
          <span
            className="prodShortLabel"
          >
            WD 2TB Elements Portable External Hard Drive - USB 3.0 
          </span>
          <span
            className="prodPrice"
          >
            R$ 
            64.00
          </span>
          <a
            className="BuyButton__BuyButtonContainer-sc-1fz6kqp-0 XtTgO"
          >
            Ver Produto
          </a>
        </div>
      </div>
      <div
        className="Product__ProductContainer-ih45b-0 iRmchH"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="prodImg"
        />
        <div
          className="prodInfoContainer"
        >
          <span
            className="prodShortLabel"
          >
            SanDisk SSD PLUS 1TB Internal SSD - SATA III 6 Gb/s
          </span>
          <span
            className="prodPrice"
          >
            R$ 
            109.00
          </span>
          <a
            className="BuyButton__BuyButtonContainer-sc-1fz6kqp-0 XtTgO"
          >
            Ver Produto
          </a>
        </div>
      </div>
      <div
        className="Product__ProductContainer-ih45b-0 lftYPQ"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="prodImg"
        />
        <div
          className="prodInfoContainer"
        >
          <span
            className="prodShortLabel"
          >
            Silicon Power 256GB SSD 3D NAND A55 SLC Cache Performance Boost S...
          </span>
          <span
            className="prodPrice"
          >
            R$ 
            109.00
          </span>
          <a
            className="BuyButton__BuyButtonContainer-sc-1fz6kqp-0 XtTgO"
          >
            Ver Produto
          </a>
        </div>
      </div>
      <div
        className="Product__ProductContainer-ih45b-0 huyKpF"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="prodImg"
        />
        <div
          className="prodInfoContainer"
        >
          <span
            className="prodShortLabel"
          >
            WD 4TB Gaming Drive Works with Playstation 4 Portable External Ha...
          </span>
          <span
            className="prodPrice"
          >
            R$ 
            114.00
          </span>
          <a
            className="BuyButton__BuyButtonContainer-sc-1fz6kqp-0 XtTgO"
          >
            Ver Produto
          </a>
        </div>
      </div>
      <div
        className="Product__ProductContainer-ih45b-0 iLUnCv"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="prodImg"
        />
        <div
          className="prodInfoContainer"
        >
          <span
            className="prodShortLabel"
          >
            Acer SB220Q bi 21.5 inches Full HD (1920 x 1080) IPS Ultra-Thin
          </span>
          <span
            className="prodPrice"
          >
            R$ 
            599.00
          </span>
          <a
            className="BuyButton__BuyButtonContainer-sc-1fz6kqp-0 XtTgO"
          >
            Ver Produto
          </a>
        </div>
      </div>
      <div
        className="Product__ProductContainer-ih45b-0 fYomlo"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="prodImg"
        />
        <div
          className="prodInfoContainer"
        >
          <span
            className="prodShortLabel"
          >
            Samsung 49-Inch CHG90 144Hz Curved Gaming Monitor (LC49HG90DMNXZA...
          </span>
          <span
            className="prodPrice"
          >
            R$ 
            999.99
          </span>
          <a
            className="BuyButton__BuyButtonContainer-sc-1fz6kqp-0 XtTgO"
          >
            Ver Produto
          </a>
        </div>
      </div>
      <div
        className="Product__ProductContainer-ih45b-0 ioLRXx"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="prodImg"
        />
        <div
          className="prodInfoContainer"
        >
          <span
            className="prodShortLabel"
          >
            BIYLACLESEN Women's 3-in-1 Snowboard Jacket Winter Coats
          </span>
          <span
            className="prodPrice"
          >
            R$ 
            56.99
          </span>
          <a
            className="BuyButton__BuyButtonContainer-sc-1fz6kqp-0 XtTgO"
          >
            Ver Produto
          </a>
        </div>
      </div>
      <div
        className="Product__ProductContainer-ih45b-0 ipLHab"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="prodImg"
        />
        <div
          className="prodInfoContainer"
        >
          <span
            className="prodShortLabel"
          >
            Lock and Love Women's Removable Hooded Faux Leather Moto Biker Ja...
          </span>
          <span
            className="prodPrice"
          >
            R$ 
            29.95
          </span>
          <a
            className="BuyButton__BuyButtonContainer-sc-1fz6kqp-0 XtTgO"
          >
            Ver Produto
          </a>
        </div>
      </div>
      <div
        className="Product__ProductContainer-ih45b-0 dYcLzK"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="prodImg"
        />
        <div
          className="prodInfoContainer"
        >
          <span
            className="prodShortLabel"
          >
            Rain Jacket Women Windbreaker Striped Climbing Raincoats
          </span>
          <span
            className="prodPrice"
          >
            R$ 
            39.99
          </span>
          <a
            className="BuyButton__BuyButtonContainer-sc-1fz6kqp-0 XtTgO"
          >
            Ver Produto
          </a>
        </div>
      </div>
      <div
        className="Product__ProductContainer-ih45b-0 IwpDE"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="prodImg"
        />
        <div
          className="prodInfoContainer"
        >
          <span
            className="prodShortLabel"
          >
            MBJ Women's Solid Short Sleeve Boat Neck V 
          </span>
          <span
            className="prodPrice"
          >
            R$ 
            9.85
          </span>
          <a
            className="BuyButton__BuyButtonContainer-sc-1fz6kqp-0 XtTgO"
          >
            Ver Produto
          </a>
        </div>
      </div>
      <div
        className="Product__ProductContainer-ih45b-0 jGnWRg"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="prodImg"
        />
        <div
          className="prodInfoContainer"
        >
          <span
            className="prodShortLabel"
          >
            Opna Women's Short Sleeve Moisture
          </span>
          <span
            className="prodPrice"
          >
            R$ 
            7.95
          </span>
          <a
            className="BuyButton__BuyButtonContainer-sc-1fz6kqp-0 XtTgO"
          >
            Ver Produto
          </a>
        </div>
      </div>
      <div
        className="Product__ProductContainer-ih45b-0 jFzYkH"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="prodImg"
        />
        <div
          className="prodInfoContainer"
        >
          <span
            className="prodShortLabel"
          >
            DANVOUY Womens T Shirt Casual Cotton Short
          </span>
          <span
            className="prodPrice"
          >
            R$ 
            12.99
          </span>
          <a
            className="BuyButton__BuyButtonContainer-sc-1fz6kqp-0 XtTgO"
          >
            Ver Produto
          </a>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="Footer__FooterContainer-qgk6jt-0 gXkdUn"
  >
    <div>
      <i
        className="icon-facebook"
      />
      <i
        className="icon-instagram"
      />
      <i
        className="icon-twitter"
      />
      <i
        className="icon-youtube"
      />
    </div>
  </footer>
</div>
`;
